install.php

```diff
@@ -1,3 +1,40 @@
 <?php
-$_GET["mod"] = "install";
-include_once("index.php");
+
+//raise error reporting - could be interesting and this is autmomatically reset when index.php is executed
+error_reporting(E_ALL);
+
+
+/**
+* Check for common failures
+*
+* Small function to check against a number of dependencies that cause install.php to fail without showing any error details
+* @return bool check result. True if error found, false if all OK
+*/
+function checkdeps(){
+
+	$error =false;
+	$errmsg='The following issues were found while preparing to run the installation:<br/>'. PHP_EOL;
+
+	//check if composer was executed (folder ./vendor created)
+	if (!is_dir(__DIR__ . '/vendor')) {
+
+		$error = true;
+		$errmsg .= __DIR__ . '/vendor directory is missing. Did you run composer?<br/>' .PHP_EOL;
+	}
+
+	//check if smarty-directory is writable
+	if (!is_writable(__DIR__ . '/ext_inc/templates_c')) {
+		$error = true;
+		$errmsg .= __DIR__ . '/ext_inc/templates_c seems not to be writable by the PHP user. Please confirm that file and folder access rights are applied correctly<br/>' . PHP_EOL;
+	}
+
+	//check error state and return stored messages if problem found
+	if ($error) echo $errmsg;
```

Comment:
I suggest using curly braces (`{`) even in one line if statements.
The reason is to avoid bugs, once we extend this clause. A quick change then can introduce bugs, because only the first statement is executed in such a condition if no curly braces are there.

Next to this:
What do you think about return an empty string, once no error occurred and a string with the error message, if an error is there? Instead of `echo` the message here?
Then we could a) test this function (e.g. with a unit test) and b) make a proper output later on (with templates or anything).
The alternative would be to throw an Exception. Just an idea. Let me know what do you think.